xarray/core/datatree.py

```diff
@@ -1238,65 +1238,67 @@ def isomorphic(
         except (TypeError, TreeIsomorphismError):
             return False
 
-    def equals(self, other: DataTree, from_root: bool = True) -> bool:
+    def _matching(
+        self,
+        other: DataTree,
+        nodes_match: Callable[[DataTree, DataTree], bool],
+    ) -> bool:
+        if not self.isomorphic(other, from_root=True, strict_names=True):
```

Comment:
I'm wondering if there is a problem with removing the `from_root` argument but keeping it fixed to `True` inside `check_isomorphic`...

Let's imagines I `assert_equal` two subtrees that have different parents. All the nodes in the subtrees could match, but if the structures above the node where the subtree starts are not isomorphic, it will return `False`. That doesn't seem like desired behaviour to me - it leaves no way to check that the two subtrees are equal.